Source/monster.cpp

```diff
@@ -4689,41 +4691,39 @@ void FreeMonsters()
 	FreeMissiles2();
 }
 
-bool DirOK(int i, direction mdir)
+bool DirOK(int i, Direction mdir)
 {
-	int fx, fy;
 	int x, y;
 	int mcount, mi;
 
 	commitment((DWORD)i < MAXMONSTERS, i);
-	fx = monster[i].position.tile.x + offset_x[mdir];
-	fy = monster[i].position.tile.y + offset_y[mdir];
-	if (fy < 0 || fy >= MAXDUNY || fx < 0 || fx >= MAXDUNX || !PosOkMonst(i, fx, fy))
+	Point futurePosition = monster[i].position.tile + mdir;
+	if (futurePosition.y < 0 || futurePosition.y >= MAXDUNY || futurePosition.x < 0 || futurePosition.x >= MAXDUNX || !PosOkMonst(i, futurePosition.x, futurePosition.y))
 		return false;
 	if (mdir == DIR_E) {
-		if (SolidLoc(fx, fy + 1) || dFlags[fx][fy + 1] & BFLAG_MONSTLR)
+		if (SolidLoc(futurePosition.x, futurePosition.y + 1) || dFlags[futurePosition.x][futurePosition.y + 1] & BFLAG_MONSTLR)
 			return false;
 	} else if (mdir == DIR_W) {
-		if (SolidLoc(fx + 1, fy) || dFlags[fx + 1][fy] & BFLAG_MONSTLR)
+		if (SolidLoc(futurePosition.x + 1, futurePosition.y) || dFlags[futurePosition.x + 1][futurePosition.y] & BFLAG_MONSTLR)
 			return false;
 	} else if (mdir == DIR_N) {
-		if (SolidLoc(fx + 1, fy) || SolidLoc(fx, fy + 1))
+		if (SolidLoc(futurePosition.x + 1, futurePosition.y) || SolidLoc(futurePosition.x, futurePosition.y + 1))
 			return false;
 	} else if (mdir == DIR_S)
-		if (SolidLoc(fx - 1, fy) || SolidLoc(fx, fy - 1))
+		if (SolidLoc(futurePosition.x - 1, futurePosition.y) || SolidLoc(futurePosition.x, futurePosition.y - 1))
 			return false;

```

Comment:
Would this be doable as `futurePosition + mdir` now?

Reply:
I'm touching this function in a subsequent PR so this will be applied there.